### DISCUSSION TITLE
How to deploy wasm component without down time

### DISCUSSION THREAD

**Discussion by goldenratio** [spinframework/spin#2379] [General]
- Let's assume I have 5 WASM components, each have there  own routes
- I want to update one component and deploy it without restarting the server

Any guidance on this is appreciated.

**Comment by lann** [comment]
`spin up` is designed primarily as a development tool; it does not directly support this kind of advanced deployment feature. There are a number of strategies for zero-downtime deployments involving proxies such as the Kubernetes-based [SpinKube](https://www.spinkube.dev/).

**Reply by goldenratio** [reply]
thank you!
